File: Trabalho_Final/TF/Final/src/tf/Repositorio.java
package tf;

import java.io.IOException;
import java.util.ArrayList;

public class Repositorio{
	
	private ArrayList<Disciplina> Disciplinas = new ArrayList<Disciplina>();
	private ArrayList<Aluno> Alunos = new ArrayList<Aluno>();
	
	public String addAlu(String nome, int matricula){
		String ss="";
		if(checkAlu(matricula) == -1) {
			Alunos.add(new Aluno(nome, matricula));
			return ss;
		}
		ss = "\n Aluno já existe \n";
		return ss;
	}
	
	public String addDisc(String id, String cod){
		String ss="";
		if(checkDisc(cod) == -1) {
			Disciplinas.add(new Disciplina(id, cod));
			return ss;
		}
		ss = "\n Disciplina já existe \n";
		return ss;
	}
	
	public String Matricular(String cod, int matricula){
		String ss ="";
		int indA = checkAlu(matricula);
		int indD = checkDisc(cod);	
		if(indA != -1) {
			if(indD != -1){
				Alunos.get(indA).Matricular(cod);
				ss = Disciplinas.get(indD).Matricular(matricula);
				return ss;
			}
			ss = "\n Disciplina não existe \n";
			return ss;
		}
		ss = "\n Aluno não existe \n";
		return ss;
	}
	
	public String Desmatricular(String cod, int matricula){
		String ss ="";
		int indA = checkAlu(matricula);
		int indD = checkDisc(cod);	
		if(indA != -1){
			if(indD != -1) {
				Alunos.get(indA).Desmatricular(cod);
				ss = Disciplinas.get(indD).Desmatricular(matricula);
				return ss;
			}
			ss = "\n Disciplina não existe \n";
			return ss;
		}
		ss = "\n Aluno não existe \n";
		return ss;
	}
	
	public String RemoverAlu(int matricula) throws IOException{
		String ss = "";
		int ind = checkAlu(matricula);
		if(ind != -1){
			ArrayList<String> aux = this.Alunos.get(ind).getDisc();			
			for(int i=0; i<Integer.valueOf(aux.size());i++){
				for(int j=0; j<Integer.valueOf(this.Disciplinas.size());j++) {
					if(aux.get(i).equals(Disciplinas.get(j).getCod())){
						Disciplinas.get(j).Desmatricular(matricula);
					}
				}
			}
			Alunos.remove(ind);
			return ss;
		}
		ss = "\n Aluno não Existe \n";
		return ss;
	}
	
	public String RemoverDisc(String cod) throws IOException{
		String ss = "";
		int ind = checkDisc(cod);
		if(ind != -1){
			ArrayList<Integer> aux = this.Disciplinas.get(ind).getAlu();		
			for(int i=0; i<Integer.valueOf(aux.size());i++){
				for(int j=0; j<Integer.valueOf(this.Alunos.size());j++) {
					if(aux.get(i) == Alunos.get(j).getMat()) {
						Alunos.get(j).Desmatricular(cod);
					}
				}
			}
			Disciplinas.remove(ind);
			return ss;
		}
		ss = "\n Disciplina não Existe \n";
		return ss;
	}

	public int checkAlu(int matricula){
		for(int i = 0; i < this.Alunos.size(); i++){
			if(matricula == Alunos.get(i).getMat())
				return i;			
		}
		return -1;		
	}
	
	public int checkDisc(String cod){
		for(int i = 0; i< this.Disciplinas.size(); i++){
			if(cod.equals(Disciplinas.get(i).getCod()))
				return i;			
		}
		return -1;		
	}
	
	public String showAlu() {
		String ss="";
		if (this.Alunos != null && !this.Alunos.isEmpty()){
			ss = "Alunos: \n";
			for(Aluno i : this.Alunos) {
				ss += "\n" + i.show() + "\n";
			}
			return ss;
		}
		ss = "\n Nenhum Aluno Cadastrado \n";
		return ss;
	}
	
	public String showDisc(){
		String ss="";
		if (this.Disciplinas != null && !this.Disciplinas.isEmpty()) {
			ss = "Disciplinas: \n";
			for(Disciplina i : this.Disciplinas) {
				ss += "\n" + i.show() + "\n";
			}
			return ss;
		}
		ss = "\n Nenhuma Disciplina Cadastrada \n";
		return ss;
	}
}

File: Trabalho_Final/TF_V1/Repositorio.py
class aluno(object):
     
    nome = None
    vecDisc = []

    def __init__(self, nome):
        self.nome = nome
        
    def Matricular(self, disc):
        for j in range(len(self.vecDisc)):
            if disc == str(self.vecDisc[j]):
                return
        self.vecDisc.append(disc)
    
    def Desmatricular(self, nome):
        for j in range(len(self.vecDisc)):
            if nome == str(self.vecDisc[j]):
                del(self.vecDisc[j])
                return

    def show(self):
        ss = self.nome + ":"
        for i in range(len(self.vecDisc)):
            ss = ss + "\n" +"\t" + str(self.vecDisc[i])
        return ss

    def getDisc(self):
        return self.vecDisc

    def getName(self):
        return self.nome



class disciplina(object):
     
    disc = None
    vecAlu = []

    def __init__(self, disc):
        self.disc = disc

    def Matricular(self, nome):
        for j in range(len(self.vecAlu)):
            if nome == str(self.vecAlu[j]):
                return
        self.vecAlu.append(nome)
    
    def Desmatricular(self, disc):
        for j in range(len(self.vecAlu)):
            if disc == str(self.vecAlu[j]):
                del(self.vecAlu[j])
                return
            
    def show(self):
        ss = self.disc + ":"
        for i in range(len(self.vecAlu)):
            ss = ss + "\n" +"\t" + str(self.vecAlu[i])
        return ss

    def getAlu(self):
        return self.vecAlu

    def getName(self):
        return self.disc



class repositorio(object):
    
    repAlu = []
    repDis = []

    def addAlu(self, nome):
        for i in range(len(self.repAlu)):
        # self.repAlu[nome] = aluno(nome)
            if nome == str(self.repAlu[i]):
                return
        self.repAlu.append(aluno(nome))

    def addDis(self, disc):
        for i in range(len(self.repDis)):
        # self.repAlu[nome] = aluno(nome)
            if disc == str(self.repDis[i]):
                return
        self.repDis.append(disciplina(disc))

    def Matricular(self, disc, nome):
       for i in range(len(self.repDis)):
           if disc == self.repDis[i].getName():
               for j in range(len(self.repAlu)):
                   if nome == str(self.repAlu[j].getName()):
                    self.repAlu[j].Matricular(disc)
                    self.repDis[i].Matricular(nome)
                    return

    def Desmatricular(self, disc, nome):
       for i in range(len(self.repDis)):
           if disc == str(self.repDis[i].getName()):
               for j in range(len(self.repAlu)):
                   if nome == self.repAlu[j].getName():
                    self.repAlu[j].Desmatricular(disc)
                    self.repDis[i].Desmatricular(nome)
                    return

    def rmAlu(self, nome):
        if nome in self.repAlu:
            indAlu = self.checkIndA(nome)
            y = self.repAlu[indAlu].getDisc()
            for i in y:
                self.repDis[i].Desmatricular(nome)
            del(self.repAlu[indAlu])

    def showAlu(self, nome):
        for i in range(len(self.repAlu)):
            if nome == str(self.repAlu[i].getName()):
                show = nome + ":"
                y = self.repAlu[i].getDisc()
                for j in range(len(y)):
                    show = show + "\n" + "\t" + str(y[j])
                return show
        return "não existe"

    def showDisc(self, nome):
        for i in range(len(self.repDis)):
            if nome == str(self.repDis[i].getName()):
                show = nome + ":"
                y = self.repDis[i].getAlu()
                for j in range(len(y)):
                    show = show + "\n" + "\t" + str(y[j])
                return show
        return "não existe"

    def showAllAlu(self):
        show = ""
        for i in self.repAlu:
            show = show +"\n"+ i.getName()
        show = show + "\n"
        return show

    def showAllDisc(self):
        show = ""
        for i in self.repDis:
            show = show +"\n"+ i.getName()
        show = show + "\n"
        return show

